acme/acme/client.py

```diff
@@ -549,14 +550,82 @@ def new_account(self, new_account):
 
         :returns: Registration Resource.
         :rtype: `.RegistrationResource`
-
         """
         response = self.net.post(self.directory['newAccount'], new_account,
             acme_version=2)
         # "Instance of 'Field' has no key/contact member" bug:
         # pylint: disable=no-member
         return self._regr_from_response(response)
 
+    def new_order(self, csr_pem):
+        """Request challenges.
+
+        :returns: List of Authorization Resources.
+        :rtype: `list` of `.AuthorizationResource`
+        """
+        csr = cryptography.x509.load_pem_x509_csr(csr_pem,
+            cryptography.hazmat.backends.default_backend())
+        san_extension = next(ext for ext in csr.extensions
+            if ext.oid == cryptography.x509.oid.ExtensionOID.SUBJECT_ALTERNATIVE_NAME)
+        dnsNames = san_extension.value.get_values_for_type(cryptography.x509.DNSName)
+
+        identifiers = []
+        for name in dnsNames:
+            identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_FQDN,
+                value=name))
+        order = messages.NewOrder(identifiers=identifiers)
+        response = self.net.post(self.directory['newOrder'], order)
+        order_response = self._order_resource_from_response(
+            response, csr_pem=csr_pem)
+        return order_response
+
+    def _order_resource_from_response(self, response, uri=None, csr_pem=None):
+        body = messages.Order.from_json(response.json())
+        authorizations = []
+        for url in body.authorizations:
+            authorizations.append(self._authzr_from_response(self.net.get(url)))
+        fullchain_pem = None
+        if body.certificate is not None:
+            certificate_response = self.net.get(body.certificate, content_type=None)
+            if certificate_response.ok:
+                fullchain_pem = certificate_response.text
+        return messages.OrderResource(
+            body=body,
+            uri=response.headers.get('Location', uri),
+            fullchain_pem=fullchain_pem,
+            authorizations=authorizations,
+            csr_pem=csr_pem)
+
+    def poll_order_and_request_issuance(self, orderr, max_time=datetime.timedelta(seconds=90)):
```

Comment:
We can keep this method if its useful for others, but we actually don't use the `poll_and_request_issuance` method in the current ACMEv1 client in Certbot. Other code handles making sure the authzs are all valid so we could get by with method that just posts to the finalize URL and polls that and avoid polling all authzs again.